ingestor/ingestor.js

```diff
@@ -0,0 +1,84 @@
+// TODOs
+// We should add the unzipper somewhere in here as well
+// Find out if we can get the relative file path instead of absolute
+// Next step after we've recursed over the archive would be to insert all the parsed data into the data base, aligning data with our mongoDB data model
+
+const fs = require('fs');
+const path = require('path');
+const { processFileController } = require('./utils.js');
+
+// Local State
+let state = {};
+state.results = [];
+state.users = [];
+state.channels = [];
+state.messages = [];
+
+const filewalker = async (dir, done) => {
+  // Read the directory (unzipped archive), this is also the recursive function.
+  await fs.readdir(dir, async function(err, list) {
+    // If at any point there is an error, invoke the callback function with the error as the argument
+    if (err) return done(err);
+
+    // Check to see if there are any items in the list
+    var pending = list.length;
+
+    // base case # 2, if there are no more items in the list, return
+    if (!pending) return done(null, users, messages, channels);
```

Comment:
Where are `users`, `messages`, and `channels` declared? Also your callback doesn't have these parameters.